test/common/http/matching/inputs_test.cc

```diff
@@ -107,6 +107,29 @@ TEST(MatchingData, HttpResponseTrailersDataInput) {
   }
 }
 
+TEST(MatchingData, HttpRequestQueryParamsDataInput) {
+  Network::ConnectionInfoSetterImpl connection_info_provider(
+      std::make_shared<Network::Address::Ipv4Instance>(80),
+      std::make_shared<Network::Address::Ipv4Instance>(80));
+  HttpMatchingDataImpl data(connection_info_provider);
+
+  {
+    HttpRequestQueryParamsDataInput input("username", false /* url_decoded */);
```

Comment:
format nit: the correct format for annotation should be `/* url_decoded=*/ false`.

Same for the test below